### PR TITLE
Fix `net_flux` calculation for shortwave solver.

### PR DESCRIPTION
**Pull request by sriharshakandala**
## Purpose 
Fix `net_flux` calculation for shortwave solver.

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/rte/shortwave2stream.jl [hunk 1]
```diff
@@ -83,21 +95,38 @@ function rte_sw_2stream_solve!(
...
+        flux_net_sw = flux_sw.flux_net
+        (; flux_up, flux_dn) = flux
         for igpt in 1:n_gpt
             ClimaComms.@threaded device for gcol in 1:ncol
                 bld_cld_mask && Optics.build_cloud_mask!(
                     view(as.cld_mask_sw, :, gcol),
                     view(as.cld_frac, :, gcol),
                     as.cld_mask_type,
                 )
-                igpt == 1 && set_flux_to_zero!(flux_sw, gcol)
```
Comment (szy21) [inline src/rte/shortwave2stream.jl:93]: Why do we not need it anymore?

**Review by szy21 — APPROVED** [review]
Thank you!

**Review by nefrathenrici — APPROVED** [review]
Looks good aside from Zhaoyi's comment. Thank you for adding the tests!

File: src/rte/shortwave2stream.jl (hunk 1 above)
Comment (sriharshakandala) [reply src/rte/shortwave2stream.jl:93]: Its handled here! https://github.com/CliMA/RRTMGP.jl/pull/435/files/e1fef53caabbd01dc672e482c1900bf05e618037#diff-69d7a939b6806cb60c5f3214b0818fbaa6465fbede867cc11b948509c089e112R115

**Pull request merged**